src/bs_size.c

```diff
@@ -224,30 +223,66 @@ static void strstrip(char *str) {
     str[i-begin] = '\0';
 }
 
-static bool multiply_size_by_unit (mpfr_t size, char *unit_str) {
+/* Helper function to convert mpf_t to mpz_t with round-toward-zero (truncate decimal) */
+static void mpz_set_f_round_zero(mpz_t rop, const mpf_t op) {
```

Comment:
Doesn't libgmp already provide `mpz_set_f` which does this exact thing? https://gmplib.org/manual/Assigning-Integers

Relying on the formatting and parsing seems hacky.